### PR TITLE
Moved compare code to its own plugin

### PR DESCRIPTION
**Pull request by abhisheksingla02**

### CODE REVIEW

File: frontend/express/public/core/events/javascripts/countly.details.views.js [hunk 1]
```diff
@@ -339,64 +178,44 @@
...
-        EventAllView.params = params;
-        this.renderWhenReady(EventAllView);
+        eventsViewWrapper.params = params;
+        this.renderWhenReady(eventsViewWrapper);
     });
 
-    app.route("/analytics/events", "all-events", function() {
+    countlyVue.container.registerTab("/analytics/events", {
+        priority: 1,
+        name: "",
```
Comment (itsiprikshit) [inline frontend/express/public/core/events/javascripts/countly.details.views.js:205]: why doesn't the tab have a name?

File: frontend/express/public/core/events/javascripts/countly.details.views.js [hunk 2]
```diff
@@ -49,191 +49,29 @@
...
+                "eventsTabs": "/analytics/events"
+            })
+        ],
         data: function() {
             return {
-                scoreTableExportSettings: {
-                    title: "CompareEvents",
-                    timeDependent: true,
-                }
+                selectedTab: (this.$route.params && this.$route.params.tab) || ""
```
[the remark this answers is not in the document]
Comment (itsiprikshit) [reply frontend/express/public/core/events/javascripts/countly.details.views.js:63]: ohhh so you first tab is /analytics/events route.

File: plugins/compare/frontend/public/javascripts/countly.views.js [hunk 3]
```diff
@@ -1,488 +1,165 @@
...
-            var appId = $(this).data("id");
+    });
 
-            if (countlyCommon.ACTIVE_APP_ID === appId) {
-                app.navigate("/", true);
-            }
+    if (countlyAuth.validateRead(FEATURE_NAME)) {
+        countlyVue.container.registerTab("/analytics/events", {
+            priority: 2,
+            name: "Compare Events",
```
Comment (itsiprikshit) [inline plugins/compare/frontend/public/javascripts/countly.views.js:157]: name should be whithout white spaces.
https://docs.google.com/document/d/1LBkmpv-HjBnYcjNf86tHiOWYEWjnxQwxQVRjXGSf5yQ/edit?pli=1#heading=h.amk244vn9egw

File: frontend/express/public/core/events/javascripts/countly.details.views.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (itsiprikshit) [reply frontend/express/public/core/events/javascripts/countly.details.views.js:63]: but still you can name it.
and you should name.
and the same name should be set here.

File: plugins/compare/frontend/public/javascripts/countly.views.js [hunk 4]
```diff
@@ -37,26 +37,18 @@
             handleCurrentChange: function(selection) {
                 var selectedEvents = [];
                 selection.forEach(function(item) {
-                    if (item.id.startsWith("[CLY]_group")) {
-                        selectedEvents.push(item.id);
-                    }
-                    else {
-                        selectedEvents.push(item.name);
```
Comment (itsiprikshit) [inline plugins/compare/frontend/public/javascripts/countly.views.js]: did you remove this intentionally?

Comment (abhisheksingla02) [reply plugins/compare/frontend/public/javascripts/countly.views.js]: yes

File: frontend/express/public/core/events/javascripts/countly.details.views.js [hunk 5]
```diff
@@ -339,64 +178,44 @@
...
+        eventsViewWrapper.params = params;
+        this.renderWhenReady(eventsViewWrapper);
     });
 
-    app.route("/analytics/events", "all-events", function() {
+    countlyVue.container.registerTab("/analytics/events", {
+        priority: 1,
+        name: "eventsDetail",
+        title: "Event Stats",
+        route: "#/" + countlyCommon.ACTIVE_APP_ID + "/analytics/events",
```
Comment (itsiprikshit) [inline frontend/express/public/core/events/javascripts/countly.details.views.js:207]: btw you can also change this route to -
 "#/" + countlyCommon.ACTIVE_APP_ID + "/analytics/events/eventsDetail",

**Pull request merged**